src/Tasks/ResourceHandling/MSBuildResXReader.cs

```diff
@@ -0,0 +1,299 @@
+            // own assembly.
+            if (aliasedTypeName.StartsWith("System.String"))
+            {
+                return StringTypeName;
+            }
+
+            // No alias found. Hope it's sufficiently complete to be resolved at runtime
+            return aliasedTypeName;
+        }
+
+        private static void ParseData(string resxFilename, bool pathsRelativeToBasePath, List<IResource> resources, Dictionary<string,string> aliases, XElement elem)
+        {
+            string name = elem.Attribute("name").Value;
+            string value;
+
+            XElement valueElement = elem.Element("value");
+            if (valueElement is null)
+            {
+                if (elem.HasElements)
+                {
+                    throw new NotImplementedException("User-facing error for bad resx that has child elements but not `value`");
+                }
+
+                value = elem.Value;
+            }
+            else
+            {
+                value = valueElement.Value;
+            }
+
+            string typename = elem.Attribute("type")?.Value;
+            string mimetype = elem.Attribute("mimetype")?.Value;
+
+            typename = GetFullTypeNameFromAlias(typename, aliases);
+
+            if (typename == StringTypeName)
+            {
+                if (mimetype == null)
+                {
+                    // If nothing is specified, or String is explicitly specified
+                    // with no mimetype: read the string from the resx and return it.
+                    resources.Add(new StringResource(name, value, resxFilename));
+                    return;
+                }
+
+                // It's a string, but it might be represented oddly.
+                // Fall through to see if one of the serializers can handle it.
+            }
+
+            if (typename.StartsWith("System.Resources.ResXFileRef", StringComparison.Ordinal)) // TODO: is this too general? Should it be OrdinalIgnoreCase?
+            {
+                AddLinkedResource(resxFilename, pathsRelativeToBasePath, resources, name, value);
+                return;
+            }
+
+            // TODO: validate typename at this point somehow to make sure it's vaguely right?
+
+            if (mimetype == null)
+            {
+                if (typename.IndexOf("System.Byte[]") != -1 && typename.IndexOf("mscorlib") != -1)
+                {
+                    byte[] byteArray = Convert.FromBase64String(value);
+
+                    // Comment and logic from https://github.com/dotnet/winforms/blob/16b192389b377c647ab3d280130781ab1a9d3385/src/System.Windows.Forms/src/System/Resources/ResXDataNode.cs#L411-L416
+                    // Handle byte[]'s, which are stored as base-64 encoded strings.
+                    // We can't hard-code byte[] type name due to version number
+                    // updates & potential whitespace issues with ResX files.
```

Comment:
🤦‍♂  This logic is pretty busted, but worst-case we parse as byte[] and flow through to app as byte[].  If they were expecting a different type (like `Sytem.Typle<System.String,System.Byte[]>, mscorlib`) they'll notice.